### ISSUE TITLE
Metabolite ID conversion gives different results, based on auto-select yes/no

### ISSUE THREAD

**Issue by DeniseSl22** [bridgedb/Goliath#7]
Hi @nunogit ;

I'm testing goliath with the list of HMDb IDs that Freddie was working on:
HMDB00112
HMDB00295
HMDB00901
HMDB00935
HMDB01275
HMDB01877
HMDB13037

When I do autoselect, I get these results:
![image](https://user-images.githubusercontent.com/26277832/54908606-96269180-4ee8-11e9-9729-130397a28814.png)

When I say I want to convert from HMDb to CHEBI, the results change (I have no mappings left):
![image](https://user-images.githubusercontent.com/26277832/54908572-8444ee80-4ee8-11e9-86eb-dd8de38dd89e.png)

And, when I change the IDs of HMDb from old to new ID structure, I get the following results:
![image](https://user-images.githubusercontent.com/26277832/54908708-d128c500-4ee8-11e9-965c-a30d4239aa5b.png)

(every ID but the last is similar... this could be due to mapping issues).

And with HMDB selected as primary database to map from (some mappings, but most do not work):
![image](https://user-images.githubusercontent.com/26277832/54908782-f4537480-4ee8-11e9-80e7-dee74426fe0b.png)

And last but not least, the ID that keeps giving "undefined" as a result, does exist in HMDB:
http://www.hmdb.ca/metabolites/HMDB0000901 . I can check if it is also part of our mapping files (could be that the webservice is still running on quite old data right?)...

**Comment by nunogit** [comment]
> And last but not least, the ID that keeps giving "undefined" as a result, does exist in HMDB:
> http://www.hmdb.ca/metabolites/HMDB0000901 . I can check if it is also part of our mapping files (could be that the webservice is still running on quite old data right?)...

Undefined does not necessarly mean the id as a source doesn't exist; It means that there is no target id;  But you raise a good point; We should make these 2 situations clear.

**Comment by DeniseSl22** [comment]
Yes true, apparently neither HMDb nor Wikidata (https://www.wikidata.org/wiki/Q47538360) have a mapping to ChEBI, so then it probably does not exist in ChEBI (yet)... So if there is no target ID, the word undefined could be replaced by "no mapping available"?

**Comment by DeniseSl22** [comment]
So I think the autoselect vs selecting a database to map from difference is a bug....
The other one (secondary and primary IDs) should be dealt with at some point (but not now, since updates from BridgeDb are needed for this).
